### PR TITLE
New parameter dryrun to specify dryrun mode

### PR DESCRIPTION
**Pull request by traylenator**
#### Pull Request (PR) description

By default the new `dryrun` parameter is `false` however if dryrun is specified
`true` then the `--dryrun` option will be added to earlyoom and earlyoom will
only log processes it is going to kill.

### CODE REVIEW

File: REFERENCE.md [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
-### earlyoom
+### `earlyoom`
 
 Installs earlyoom and configures earlyoom
 
 * **See also**
-https://github.com/rfjakob/earlyoom
-earlyoom homepage
+  * https://github.com/rfjakob/earlyoom
+    * earlyoom homepage
```
Comment (igalic) [inline REFERENCE.md:27]: this looks odd
maybe just make it a regular link:

```markdown
[earlyoom Homepage](https://github.com/rfjakob/earlyoom)
```

File: REFERENCE.md [hunk 2]
```diff
@@ -62,87 +64,87 @@ Data type: `String[1]`
...
+Default value: `'/etc/default/earlyoom'`
 
 ##### `service_enable`
 
 Data type: `Boolean`
 
 Should the earlyoom service be started.
 
-Default value: `true`
+Default value: ``true``
```
Comment (igalic) [inline REFERENCE.md:91]: this looks odd.
why are the backticks doubled?

File: manifests/config.pp [hunk 3]
```diff
@@ -96,12 +101,13 @@
     $_swap_percent,
     $_memory_size,
     $_swap_size,
+    $_dryrun,
   ]
 
-  file{$configfile:
-    ensure       => present,
+  file { $configfile:
+    ensure       => file,
```
Comment (igalic) [inline manifests/config.pp:108]: why this change?

File: manifests/config.pp [hunk 4]
```diff
@@ -96,12 +101,13 @@
...
     content      => epp('earlyoom/default.epp',{
-      'options' => $_options.delete('').join(' ')
+        'options' => $_options.delete('').join(' ')
```
Comment (igalic) [inline manifests/config.pp:110]: this looks wrong, that is four instead of two spaces

**Review by igalic — CHANGES_REQUESTED** [review]
🙋🏻‍♀️

File: manifests/config.pp (hunk 3 above)
Comment (traylenator) [reply manifests/config.pp:108]: New voxpupli-test gem has pulled in newer lints including the one that dislikes

Happy to put this on hold till #8 is merge as a more natural place for the lint fixes.

File: REFERENCE.md (hunk 1 above)
Comment (traylenator) [reply REFERENCE.md:27]: This is auto generated from:

```
# @see https://github.com/rfjakob/earlyoom earlyoom homepage
```

which seems to be correct syntax - https://www.rubydoc.info/gems/yard/file/docs/Tags.md#see

Comment (igalic) [reply REFERENCE.md:27]: i wonder if you bn sensibly do

```markdown 
@see [earlyoom Homepage](https://github.com/rfjakob/earlyoom)
```

Comment (traylenator) [reply REFERENCE.md:27]: Looking at the resulting rendering on the markdown I would say it's perfectly readable.

File: REFERENCE.md (hunk 2 above)
Comment (traylenator) [reply REFERENCE.md:91]: It does look odd I agree but again I'd say the rendering looks fine.

Looked it up,  using double ticks allows for a single tick to appear in the middle non escaped.

File: manifests/config.pp (hunk 4 above)
Comment (traylenator) [reply manifests/config.pp:110]: Agree I re-gigged it in #8 so that it behaves but I agree it looks like a linter error, will report.

File: manifests/config.pp (hunk 3 above)
Comment (igalic) [reply manifests/config.pp:108]: aye. let's wait for that then

**Review by igalic — APPROVED** [review]
👍

**Comment by traylenator** [comment]
@igalic many thanks.
